site/app/templates/polls/ViewPoll.twig

```diff
@@ -165,10 +207,15 @@
                     color: [],
                 },
             }];
+
+            // Store the position of each response for incoming websocket updates (-1, 0, 1)
+            const responseIndexMap = {};
+            let position = 0;
             const [correctColor, defaultColor] =
             [window.getComputedStyle(document.body).getPropertyValue('--histogram-correct'),
                 window.getComputedStyle(document.body).getPropertyValue('--histogram-default')];
             {% for option in poll.getOptions() %}
+                responseIndexMap["{{ option.getResponse()|replace({"\n": " ", "\r": " ", "\t": " "})|e('js') }}"] = position++;
                 data[0].x.push("{{ option.getResponse()|replace({"\n": " ", "\r": " ", "\t": " "})|e('js') }}");
```

Comment:
Let's not duplicate code here and instead assign the value of the twig expression to a constant.